examples/demo_runner.py

```diff
@@ -75,16 +76,21 @@ def make_cfg(self, settings):
                 sensor_specs.append(sensor_spec)
 
         # create agent specifications
-        agent_cfg = hsim.AgentConfiguration()
+        agent_cfg = habitat_sim.agent.AgentConfig()
         agent_cfg.sensor_specifications = sensor_specs
         agent_cfg.action_space = {
-            "move_forward": hsim.ActionSpec("moveForward", {"amount": 0.25}),
-            "look_left": hsim.ActionSpec("lookLeft", {"amount": 10}),
-            "look_right": hsim.ActionSpec("lookRight", {"amount": 10}),
+            "move_forward": habitat_sim.agent.ActionSpec(
+                "move_forward", habitat_sim.agent.ActuationSpec(amount=0.25)
+            ),
+            "turn_left": habitat_sim.agent.ActionSpec(
+                "turn_left", habitat_sim.agent.ActuationSpec(amount=10.0)
+            ),
+            "turn_right": habitat_sim.agent.ActionSpec(
+                "turn_right", habitat_sim.agent.ActuationSpec(amount=10.0)
+            ),
         }
-        sim_cfg.agents = [agent_cfg]
 
-        return sim_cfg
+        return sim_cfg, [agent_cfg]
```

Comment:
The difficulty with that is comparison of configs.  If the agent config is part of the sim config, that would imply that two sim configs with different agent configs would be different.  Since we have a custom `==` operator implemented because of shared pointers, this isn't the case as the agent config definition lives in python.

It is also nice as you can now change the agent config without the backend reloading anything.

Reply:
What would work is having a higher level config that has both a backend config (the current `sim_cfg`) and a list of `agent_cfg`.  Then the comparison would function as expected.

Reply:
Yes, agreed -- I think a higher level config perhaps general `Configuration` that includes the `SimulatorConfig` and `AgentConfig`s is a good idea.

Reply:
Added